### PR TITLE
Retry connect

### PR DESCRIPTION
**Pull request by nurse**
Retry on temporary DatabaseConnectionError.

### CODE REVIEW

**Comment by nurse** [comment]
@kamipo Updated to use exponential backoff.

File: spec/rdb_compat_backend_spec.rb [hunk 1]
```diff
@@ -376,6 +376,18 @@
...
+      context 'cannot connect' do
+        let (:config){ {url: 'mysql2://root:@nonexistent/perfectqueue_test', table: table} }
+        it 'raises Sequel::DatabaseConnectionError' do
+          allow(STDERR).to receive(:puts)
+          d = Backend::RDBCompatBackend.new(client, config)
+          slept = 0
+          expect(d).to receive(:sleep).exactly(9).times{|n| slept += n }
+          expect(d.db).to receive(:connect).exactly(10).times.and_call_original
+          expect{ d.__send__(:connect){ d.db.run('SELECT 1;') } }.to raise_error(Sequel::DatabaseConnectionError)
+          expect(slept).to eq(1022)
```
Comment (kamipo) [inline spec/rdb_compat_backend_spec.rb]: Probably we need to configurable to max_retry count.
The default `MAX_RETRY ` is 10, 1022 secs (17 mins) seems too long.

Comment (nurse) [inline spec/rdb_compat_backend_spec.rb]: After some consideration, I came up with Timeout.timeout...

**Comment by kamipo** [comment]
LGTM!!

**Pull request merged**